src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ReadEnvelopesControllerTest.java

```diff
@@ -128,6 +128,14 @@ public void should_not_accept_invalid_statuses_when_reading_envelopes_by_status(
     private List<EnvelopeResponse> envelopesInDb() throws Exception {
         Envelope envelope = EnvelopeCreator.envelope("BULKSCAN", Status.PROCESSED);
         envelope.setZipFileName("7_24-06-2018-00-00-00.zip"); // matches expected response file
+        // todo: make this better :(
+        envelope.getScannableItems().forEach(item -> {
```

Comment:
Why do we set DCNs like this?

Reply:
Please review full PR and then make questions/assumptions.

This is because we now make DCN unique and `EnvelopeCreator.envelope` is used across many tests. Since it is dynamic and verification part is still static, for integration test it is fine to flip back to previous value which makes single test case pass. That single test case verifies full response object returned by controller giving the data (which is presumably returned from db) is given.

P.S. The change in `EnvelopeCreator.envelope` can be inspected further below in this PR

Reply:
By looking at the code it's very hard to figure out where these values come from, what they mean and why we're setting them. How about feeding `expectedEnvelopes()` with generated DCNs instead, so that the expected JSON would reflect the actual unchanged envelopes? Would require some dynamic replacements in the JSON file.

Reply:
exactly, extra replacements for json files. i'd rather keep it as separate pr hence the `//todo` comment

Reply:
yeah can be done as a separate PR